aiida_quantumespresso/workflows/pw/relax.py

```diff
@@ -96,6 +97,60 @@ def define(cls, spec):
             help='The successfully relaxed structure, unless `relax_type is RelaxType.NONE`.')
         # yapf: enable
 
+    @classmethod
+    def get_default_protocol(cls):
+        """Return the default protocol.
+
+        :return: name of the default protocol.
+        """
+        from aiida_quantumespresso.workflows.protocols.utils import get_default_protocol
+        return get_default_protocol(cls)
+
+    @classmethod
+    def get_available_protocols(cls):
+        """Return the dictionary of available protocols.
+
+        :return: dictionary of available protocols, where each key is a protocol and value is another dictionary that
+            contains at least the key `description` and optionally other keys with supplementary information.
+        """
+        from aiida_quantumespresso.workflows.protocols.utils import get_available_protocols
+        return get_available_protocols(cls)
+
+    @classmethod
+    def get_builder_from_protocol(
+        cls, code, structure, protocol=None, overrides=None, relax_type=None, **kwargs
```

Comment:
It is a good point. I was struggling with the `overrides` and `kwargs` as well. The reason why I didn't want to start copying keywords of subprocesses literally, is that then you will have to start doing this recursively, bubbling it all the way up to the top. I don't think that will be tractable in the long run. The problem is that it is not quite clear what the difference between overrides and kwargs is. Would be good to discuss this and see if we can come up with a consistent and intelligible model

Reply:
I am pretty sure that I have a working model now. The `overrides` are passed into the `get_inputs_from_protocol` call so only affect the inputs of the protocol. They should therefore have the exact same namespace structuring as the workchain and this the protocol. The `kwargs` instead are passed to the `get_builder_from_protocol` of any of the subprocesses that it runs.

The problem I see, that I just thought of now, is that I have removed the `**kwargs` from the `PwBaseWorkChain.get_builder_from_protocol` method since it doesn't take any (as it doesn't have any sub workflows, although it has the `PwCalculation` as subprocess, but that doesn't have a protocol generator). However, I think this will fail if `PwBandsWorkChain.get_builder_from_protocol` gets the `relax_type` keyword argument. This will be passed on fine to the relax workchain generator, but _also_ to the base generator, which will then fail with an unknown keyword argument. Maybe I should just add `**kwargs` to the base input generator method again, even though it doesn't do anything with it. What do you think?

Reply:
Just confirmed that my suspicion was right and `PwBaseWorkChain.get_builder_from_protocol` also needs the `**kwargs` input even if it doesn't use it. It is okay to add it I think, this occurs often in Python methods. I have added a test to make sure this works.